lib/features/restauran_tour/view/restaurant_detail_screen.dart

```diff
@@ -0,0 +1,204 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:restaurantour/features/restauran_tour/model/restaurant.dart';
+import 'package:restaurantour/features/restauran_tour/view_model/restaurants_view_model.dart';
+import 'package:restaurantour/theme/app_color.dart';
+import 'package:restaurantour/widgets/open_status.dart';
+import 'package:restaurantour/widgets/profile_avatar.dart';
+import 'package:restaurantour/widgets/rating.dart';
+
+class ResturantDetailScreen extends StatelessWidget {
+  const ResturantDetailScreen({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    RestaurantsViewModel viewModel = context.watch<RestaurantsViewModel>();
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(
+          viewModel.selectedRestaurant.name ?? "",
+          style: Theme.of(context).textTheme.headline6,
+          overflow: TextOverflow.ellipsis,
+        ),
+        actions: [
+          InkWell(
+            onTap: () {
+              viewModel
+                  .addOrRemoveFavorite(viewModel.selectedRestaurant.id ?? "");
+            },
+            child: Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 8.0),
+              child: viewModel.isFavorite()
+                  ? const Icon(Icons.favorite)
+                  : const Icon(Icons.favorite_border),
+            ),
+          ),
+        ],
+      ),
+      body: SingleChildScrollView(
+        child: Column(
+          children: [
+            _imageDisplay(context, viewModel.selectedRestaurant.heroImage),
+            _detailContent(context, viewModel.selectedRestaurant),
+          ],
+        ),
+      ),
```

Comment:
suggestion: Instead of using 2 nested widgets take advantage of the more accurated

```suggestion
    body: ListView(
        children: [
          _imageDisplay(context, viewModel.selectedRestaurant.heroImage),
          _detailContent(context, viewModel.selectedRestaurant),
        ],
      ),
```